src/Keboola/DbExtractor/Extractor/Snowflake.php

```diff
@@ -113,13 +110,56 @@ public function export(array $table): array
         $outputTable = $table['outputTable'];
 
         $this->logger->info('Exporting to ' . $outputTable);
-
+        $maxValue = null;
+        if (isset($table['table']) && isset($this->incrementalFetching)) {
+            $maxValue = $this->getMaxOfIncrementalFetchingColumn($table['table']);
+        }
         $rowCount = $this->exportAndDownload($table);
 
-        return [
-            'outputTable'=> $outputTable,
+        $output = [
+            'outputTable' => $outputTable,
             'rows' => $rowCount,
         ];
+        // output state
+        if ($maxValue) {
+            $output['state']['lastFetchedRow'] = $maxValue;
+        }
+
+        return $output;
+    }
+
+    public function getMaxOfIncrementalFetchingColumn(array $table): ?string
+    {
+
+        if (isset($this->incrementalFetching['limit']) && $this->incrementalFetching['limit'] > 0) {
+            $fullsql = sprintf(
+                'SELECT %s FROM %s.%s',
+                $this->db->quoteIdentifier($this->incrementalFetching['column']),
+                $this->db->quoteIdentifier($table['schema']),
+                $this->db->quoteIdentifier($table['tableName'])
+            );
+
+            $fullsql .= $this->createIncrementalAddon();
+
+            $fullsql .= sprintf(
+                ' LIMIT %s OFFSET %s',
+                $this->incrementalFetching['limit'],
+                $this->incrementalFetching['limit'] - 1
```

Comment:
This is a dangerous assumption.  We can't assume that the limit was the same the last time it was executed.

Reply:
treba nechapu co chces tady delat

Reply:
sry - now this solution is better

i need get last row from limiting incremental fetching for new state

Reply:
How can this work without Ordering?